hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/RMWebServices.java

```diff
@@ -734,7 +739,80 @@ public Response updateAppState(AppState targetState,
 
     return Response.status(Status.OK).entity(ret).build();
   }
-  
+
+  // Copy-paste job from updateAppQueue() above . Almost exactly the same pattern as YARN-1702.
+  // can't return POJO because we can't control the status code
+  // it's always set to 200 when we need to allow it to be set
+  // to 202
+  @PUT
+  @Path("/node/{nodeid}/vcores")
+  @Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+  @Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+  public Response updateNodeCapacity(NodeCapacity targetCapacity,
+      @Context HttpServletRequest hsr, @PathParam("nodeid") String nodeId)
+      throws AuthorizationException, YarnException, InterruptedException,
+      IOException {
+
+    init();
+    UserGroupInformation callerUGI = getCallerUserGroupInformation(hsr, true);
+    if (callerUGI == null) {
+      String msg = "Unable to obtain user name, user not authenticated";
+      throw new AuthorizationException(msg);
+    }
+
+    if (UserGroupInformation.isSecurityEnabled() && isStaticUser(callerUGI)) {
+      String msg = "The default static user cannot carry out this operation.";
+      return Response.status(Status.FORBIDDEN).entity(msg).build();
+    }
+
+    String userName = callerUGI.getUserName();
+    RMNode node = null;
+    NodeId nId = ConverterUtils.toNodeId(nodeId);
+    node = rm.getRMContext().getRMNodes().get(nId);
+    if(node == null) {
+      RMAuditLogger.logFailure(userName, AuditConstants.UPDATE_NODE_RESOURCES,
+        "UNKNOWN", "RMWebService", "Trying to change unknown node " + nodeId);
+      throw new BadRequestException("Couldn't find node " + nodeId);
+    }
+
+    //TODO(raviprak) : See if you want to change memory too
+    int memory = node.getTotalCapability().getMemory();
+    int vCores = targetCapacity.getVCores();
+    Resource resource = Resource.newInstance(memory, vCores);
+    ResourceOption resourceOption = ResourceOption.newInstance(resource, -1);
+    Map<NodeId, ResourceOption> map = new HashMap<>();
+    map.put(node.getNodeID(), resourceOption);
+    final UpdateNodeResourceRequest req =
```

Comment:
FYI not sure if you wanted to put this in 2.4.1, but the client side has a bug in that version that makes the PB building crash always.   See https://github.com/VertiCloud/alti_node_decomissioner/blob/master/src/main/java/com/altiscale/decommissioner/Decommissioner.java#L240 

It's fixed somewhere before 2.7.1 though.